### PR TITLE
[docs] Add Optim to list of solvers

### PR DESCRIPTION
**Pull request by blegat**
Preview https://jump-dev.github.io/JuMP.jl/previews/PR3624

## Basic

 - [x] Check that the solver is a registered Julia package
 - [x] Check that the MOI wrapper is part of the latest release
 - [x] Check that the solver supports the long-term support release of Julia
 - [x] Check that the solver has a MathOptInterface wrapper
 - [x] Check that the tests call `MOI.Test.runtests`. Some test excludes are
       permissible, but the reason for skipping a particular test should be
       documented.
 - [x] Check that the README and/or documentation provides an example of how to
       use the solver with JuMP https://github.com/JuliaNLSolvers/Optim.jl/pull/1059

## Documentation

 - [x] Add a new row to the table in `docs/src/installation.md`

## Optional

 - [x] Add package metadata to `docs/packages.toml`

cc @pkofod

### CODE REVIEW

File: docs/packages.toml [hunk 1]
```diff
@@ -160,6 +160,9 @@
     rev = "v1.1.0"
 # [NLopt]                   # Hard-coded in the file docs/src/packages/NLopt.md
 #     user = "JuliaOpt"
+[Optim]
+    user = "JuliaNLSolvers"
+    rev = "75ff19c14e0539f8953bd119e487a2b209987a65"
```
Comment (odow) [inline docs/packages.toml:165]:
```suggestion
    rev = "v1.9.1"
```

**Comment by odow** [comment]
@blegat the issue now is that the Optim README doesn't conform to our needs: https://jump.dev/JuMP.jl/previews/PR3624/packages/Optim/

image

Perhaps we could just omit the README for now?

**Comment by pkofod** [comment]
> the issue now is that the Optim README doesn't conform to our needs:

where can I find the requirements?

**Comment by odow** [comment]
It needs to be documenter markdown, not GitHub markdown :smile:

You can see from the image above. The links don't work, and it needs to be ` ```math ` instead of `$$`. It can be painful to get right. But I can take a look if you don't mind me making seeming pointless changes to the README just so JuMP can embed the README in our docs.

**Comment by pkofod** [comment]
> It needs to be documenter markdown, not GitHub markdown 😄
> 
> You can see from the image above. The links don't work, and it needs to be `` ```math `` instead of `$$`. It can be painful to get right. But I can take a look if you don't mind me making seeming pointless changes to the README just so JuMP can embed the README in our docs.

If you have time! Otherwise I'll try to get to it early next week.

**Comment by odow** [comment]
See https://github.com/JuliaNLSolvers/Optim.jl/pull/1076

File: docs/packages.toml [hunk 2]
```diff
@@ -160,6 +160,9 @@
     rev = "v1.1.0"
 # [NLopt]                   # Hard-coded in the file docs/src/packages/NLopt.md
 #     user = "JuliaOpt"
+[Optim]
+    user = "JuliaNLSolvers"
+    rev = "7665d5a624a35167ce7024f5559ca174b01cc6f4"
```
Comment (odow) [inline docs/packages.toml:165]: Needs commit to `master`

**Comment by pkofod** [comment]
https://github.com/JuliaRegistries/General/pull/100311

File: docs/packages.toml (hunk 2 above)
Comment (odow) [inline docs/packages.toml:165]:
```suggestion
    rev = "v1.9.2"
```

File: docs/styles/Vocab/JuMP-Vocab/accept.txt [hunk 3]
```diff
@@ -227,6 +228,7 @@ Nemhauser
 Nemirovski
 Pacaud
 Peng
+pkofod
```
Comment (odow) [inline docs/styles/Vocab/JuMP-Vocab/accept.txt:231]:
```suggestion
```

**Comment by odow** [comment]
https://jump.dev/JuMP.jl/previews/PR3624/packages/Optim/

**Pull request merged**